Assets/Scripts/UI/HUDManager.cs

```diff
@@ -0,0 +1,68 @@
+using FMS.TapperRedone.Inventory;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class HUDManager : MonoBehaviour
+{
+
+    [SerializeField] private Image drinkHUDImage;        // Drag your DrinkHUDImage here in the inspector
+    [SerializeField] private Sprite defaultSprite;       // Sprite to show when no drink is held
+
+    private PlayerInventory playerInventory;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        // Assuming PlayerInventory is attached to the player
+        playerInventory = FindObjectOfType<PlayerInventory>();
```

Comment:
Incorrect comment - you are searching across all game objects for it, attached to player or not doesn't matter here. Consider passing the player or the PlayerInventory in editor, rather than searching across all objects for it.

Reply:
will look into that, that does sound bettter to get it directly

Reply:
Player inventory is now serializable field, added player gameobject to it in inspector